Backend/starter_project_zsilentcoders/BlogApp3/BlogApp.Application/Features/Comments/CQRS/Handlers/GetCommentDetailQueryHandler.cs

```diff
@@ -0,0 +1,30 @@
+using AutoMapper;
+using BlogApp.Application.Contracts.Persistence;
+using BlogApp.Application.Features._Indices.DTOs;
+using BlogApp.Application.Features.Comments.CQRS.Queries;
+using BlogApp.Application.Features.Comments.DTOs;
+using BlogApp.Application.Responses;
+using MediatR;
+
+namespace BlogApp.Application.Features.Comments.CQRS.Handlers;
+
+public class GetCommentDetailQueryHandler : IRequestHandler<GetCommentDetailQuery, Result<CommentDto>>
+{
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
+
+        public GetCommentDetailQueryHandler(IUnitOfWork unitOfWork, IMapper mapper)
+        {
+            _unitOfWork = unitOfWork;
+            _mapper = mapper;
+        }
+        public async Task<Result<CommentDto?>> Handle(GetCommentDetailQuery request, CancellationToken cancellationToken)
+        {
+            var comment = await _unitOfWork._CommentRepository.Get(request.Id);
```

Comment:
Handle the case whether the comment found or not

Reply:
We don't have to check whether the comment was found since if there is no comment, it can return null, which is valid.

Reply:
ok , i get you, so now i make sure to be the id valid